### PR TITLE
Add `beforeScreenshotCallback` to `SentryFlutterOptions`

### LINKED ISSUE #1540: Add `before-capture-callback` and debouncing for screenshots/view hierarchy
### Description

Right now there's little control over when screenshots or view hierarchies are captured. This can have a negative impact on the app performance itself (capturing many screenshots, contribution to quota) as well as the usability of the feature itself (no control when a screenshot should be taken or not).

- backlink https://github.com/getsentry/team-mobile/issues/135

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Add `beforeScreenshotCallback` to `SentryFlutterOptions`

## :bulb: Motivation and Context

Closes #1540

## :green_heart: How did you test it?

- Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by buenaflor — APPROVED** [review]
👍

**Comment by Juliocsreis** [comment]
Was looking for this, thanks!  When it will be released ?

**Comment by buenaflor** [comment]
@Juliocsreis hey we'll do a release soon :)

**Comment by buenaflor** [comment]
@Juliocsreis this feature is now included in 7.15.0

File: flutter/lib/src/sentry_flutter_options.dart [hunk 1]
```diff
@@ -289,3 +297,10 @@ class SentryFlutterOptions extends SentryOptions {
   /// The [navigatorKey] is used to add information of the currently used locale to the contexts.
   GlobalKey<NavigatorState>? navigatorKey;
 }
+
+/// Callback being executed in [ScreenshotEventProcessor], deciding if a
+/// screenshot should be recorded and attached.
+typedef BeforeScreenshotCallback = FutureOr<bool> Function(
+  SentryEvent event, {
+  Hint? hint,
```
Comment (buenaflor) [inline flutter/lib/src/sentry_flutter_options.dart:305]: @denrase question: should hint be nullable here? afaik we removed nullable hints

Comment (buenaflor) [reply flutter/lib/src/sentry_flutter_options.dart:305]: if we don't need it we can update it in the v8 branch after I merged main into it

**Comment by Juliocsreis** [comment]
> @Juliocsreis this feature is now included in 7.15.0

Thank you very much 😄

File: flutter/lib/src/sentry_flutter_options.dart (hunk 1 above)
Comment (denrase) [reply flutter/lib/src/sentry_flutter_options.dart:305]: Lets check this in a separate issue.

**Pull request merged**